assets/js/googlesitekit/modules/datastore/modules.js

```diff
@@ -206,11 +297,31 @@ const baseSelectors = {
 	 * @param {Object} state Data store's state.
 	 * @return {(Object|undefined)} Modules available on the site.
 	 */
-	getModules( state ) {
+	getModules: createRegistrySelector( ( select ) => ( state ) => {
 		const { modules } = state;
 
-		return modules;
-	},
+		// Return `undefined` if modules haven't been loaded yet.
+		if ( modules === undefined ) {
+			return undefined;
+		}
+
+		const registryKey = select( CORE_SITE_STORE_NAME ).getRegistryKey();
+		const sortedModules = sortBy( modules, [ ( { order } ) => order ] );
+		const mappedModules = Object.values( sortedModules ).map( ( module ) => {
+			const moduleWithComponent = { ...module };
+			if ( ModuleComponents[ registryKey ] ) {
+				// If there is a settingsComponent that was passed use it, otherwise set to the default.
+				if ( ModuleComponents[ registryKey ][ module.slug ] ) {
+					moduleWithComponent.settingsComponent = ModuleComponents[ registryKey ][ module.slug ];
+				} else {
+					moduleWithComponent.settingsComponent = DefaultModuleSettings;
+				}
+			}
+
+			return moduleWithComponent;
+		} );
+		return keyBy( mappedModules, 'slug' );
```

Comment:
Using this on its own is actually a bit more readable, for what it's worth. Maybe I just don't like lodash. 😅

Reply:
We actually had some discussions way back about using native JS over `lodash` where reasonably possible, so I'd prefer if we stuck with that approach. The sorting the `core/widgets` store currently has should be easy enough to use instead of `sortBy`. And `keyBy` we should be able to easily replace too?